src/net/posix-stack.cc

```diff
@@ -325,6 +325,21 @@ posix_data_source_impl::get() {
     });
 }
 
+future<size_t, temporary_buffer<char>>
+posix_data_source_impl::get_direct(char* buf, size_t size) {
+    if (size > _buf_size / 2) {
```

Comment:
I think I cannot move it up, because this is the special case to reduce syscall for posix sockets, which is not general to other concrete `data_source_impl` classes such as `native_data_source_impl`.